src/pytesmo/validation_framework/adapters.py

```diff
@@ -309,14 +314,18 @@ def _adapt(self, data):
             else:
                 raise ValueError('"{}" is not a valid operator'.format(op))
 
-            new_mask = operator(data[column_name], threshold)
+            if self.ignore_nans:
+                new_mask = operator(data[column_name], threshold) | np.isnan(
+                    data[column_name])
+            else:
+                new_mask = operator(data[column_name], threshold)
 
             if mask is not None:
                 mask = mask & new_mask
             else:
                 mask = new_mask
 
-        return data[mask]
+        return data[mask].dropna(how="all")
```

Comment:
What's the reason for adding the `dropna` here?

Reply:
With the `ignore_nans` option active, the dataframe can be returned with all-nans rows. I think it is quite specific to the SMOSL2 data, where there are a LOT of NaNs in the flagging fields, and a lot of observation with no soil moisture (NaN) and some type of flag value.

In the latter case it can happen that the NaN soil moisture is kept. `dropna` just eliminates these occurrences

Reply:
Sounds okay to me. I will merge it in then. Do you need a new release?

Reply:
I guess it would be best, but if you prefer to wait it should work too.